lib/srv/forward/sshserver.go

```diff
@@ -620,6 +678,60 @@ func (s *Server) Serve() {
 	go s.handleConnection(ctx, chans, reqs)
 }
 
+func (s Server) setupTunnelForOpenSSHEphemeralNode(ctx context.Context) (ssh.Signer, error) {
+	if s.targetServer.GetCloudMetadata() == nil || s.targetServer.GetCloudMetadata().AWS == nil {
+		return nil, trace.BadParameter("missing aws cloud metadata")
+	}
+	awsInfo := s.targetServer.GetCloudMetadata().AWS
+
+	issuer, err := awsoidc.IssuerFromPublicAddress(s.proxyPublicAddress)
```

Comment:
We are threading the `s.ProxyPublicAddress` through a *lot* of API calls, but I don't think this is necessary. Can we call `s.authClient.GetProxies()` right here and implement similar logic to what we already do in several places, like this one:

https://github.com/gravitational/teleport/blob/befa606ed60f1611a9326f18132767181f3bdc62/tool/tctl/common/auth_command.go#L1026-L1039

I think this would make the code much easier to follow.

Reply:
We can :+1: 
I looked into this but then it didn't felt like we should do an API call for a value that rarely (never?) changes

I can change this if you feel strong about it, tho

Reply:
Yep, I'd be pretty happy with having this changed 🙏 If the extra API call becomes a performance consideration, we can safely cache it, but I don't think we should worry about that bit right now. It would simplify the data flow quite a bit, which I consider a win, given that our architecture is already fairly complex.

Reply:
Can you please take another look?

Reply:
Looks great, thank you!